lib/SILOptimizer/SILCombiner/SILCombinerApplyVisitors.cpp

```diff
@@ -1095,7 +1232,11 @@ SILInstruction *SILCombiner::visitApplyInst(ApplyInst *AI) {
   // (apply (witness_method)) -> propagate information about
   // a concrete type from init_existential_addr or init_existential_ref.
   if (auto *WMI = dyn_cast<WitnessMethodInst>(AI->getCallee())) {
-    propagateConcreteTypeOfInitExistential(AI, WMI);
+    if (!propagateConcreteTypeOfInitExistential(AI, WMI) &&
+        (WMI = dyn_cast<WitnessMethodInst>(AI->getCallee()))) {
+      // Propagate concrete type from ProtocolConformanceAnalysis.
+      propagateSoleConformingType(AI, WMI);
```

Comment:
I think if propagateConcreteTypeOfInitExistential() returned a boolean indicating if the WMI was replaced, you would only call `propagateSoleConformingType()` if it returned false, and not have to worry about getting the WMI out again

Reply:
OK. Will create a new PR with propagateConcreteTypeOfInitExistential() returning true/false.